Reemit.Common.UnitTests/RangeMappedTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿namespace Reemit.Common.UnitTests;
+
+public sealed class RangeMappedTests
+{
+    [Fact]
+    public void With_Called_ConstructsWithNewValue()
+    {
+        // Arrange
+        var rangeMapped = new RangeMapped<int>(0x52, 0x30, 0x1592);
+
+        // Act
+        var actualRangeMapped = rangeMapped.With((uint)0xdeadbeef);
```

Comment:
I'm always a little bit skeptical in situations like this, because one wrong move and the value may not be a `uint` as expected. You're right though, it is redundant. Changed it, and added an extra (somewhat unnecessary) check to quell my fears.